src/conways-game-of-life/index.ts

```diff
@@ -0,0 +1,81 @@
+const updateCellStatus = (x: number, y: number, currentCellStatus: number, neighbourSum: number) => {
+    if (currentCellStatus === 1 && neighbourSum > 3) {
+        return 0;
+    }
+    else if (currentCellStatus === 1 && neighbourSum < 2) {
+        return 0;
+    }
+    else if (currentCellStatus === 0 && neighbourSum === 3) {
+        return 1;
+    }
+    else {
+        return currentCellStatus;
+    }
+} //Apply the conditions for updating the status of a cell 
+
+const calculateNeighbourSum = (grid: number[][], x: number, y: number) => {
+    let toBeSummed: number[] = [
+        grid[y-1]?.[x+1],
+        grid[y]?.[x+1],
+        grid[y+1]?.[x+1],
+        grid[y+1]?.[x],
+        grid[y+1]?.[x-1],
+        grid[y]?.[x-1],
+        grid[y-1]?.[x-1],
+        grid[y-1]?.[x]   
+    ];
```

Comment:
Even though your logic here makes sense, you used hardcoded values to find the neighbours. You can achieve the save result by traversing the grid from `i = x -1 to x + 1` and `j = y - 1 to y + 1`. 

On the same note, you should avoid going out of bounds. I mean, only read the values inside the grid.

Let me know if you need clarification on this.

Reply:
Ok. I will implement a new solution without going out of bounds, but is there a reason for not going out of bounds? Is it less efficient ? Or create bugs? When I was thinking about that, nothing came to my mind.
Regarding the hardcoded values, why mine is hardcoded? I might need some clarification on that.

Reply:
Done!

Reply:
There are a couple of reasons:

1. Going out of bounds is not always possible (in some cases it might raise an exception)
2. You are doing unnecessary operations
* You always get 8 cells - Keep mind that in some cases you have deal with 5 cell (edges), in other cases you have only 3 cells (corners) to deal with but you always gets 8;
* Once you get all 8 cells, you filter the ones you really want by going through all 8 cells again and checking if they are undefined or not;
* Finally, you do math with the filtered cells (8, 5, or 3).

How I read that, assuming a 5x5 grid:
```
([Read cells] + [Filters Cells] + [Sum cells]) * (5 X 5) = Operations per integration
-----
([8] + [8] + [8 or 5 or 3]) * (5 X 5) = 544
```
If you make the math a 5x5 grid is not a problem. In a 100x100 grid the number grows exponentially (240,000). We can be more efficient by going strait to the cell we want.

_We can talk more about the math if you want._

<img width="881" alt="image" src="https://github.com/user-attachments/assets/86dd4edd-8327-4485-a27a-c16bbc8275f9">


3. You are learning, and the thinking process is part of the exercise. 

> Regarding the hardcoded values, why mine is hardcoded?

You always get 8 cells, even when you don't need 8.